tests/DMSDocumentCartCheckoutPageTest.php

```diff
@@ -0,0 +1,173 @@
+        $this->assertInstanceOf('FieldList', $form->Fields());
+    }
+
+    /**
+     * Tests if the DMSDocumentRequestForm is extensible
+     */
+    public function testDMSDocumentRequestFormIsExtensible()
+    {
+        $controller = DMSDocumentCartCheckoutPage_Controller::create();
+        $form = $controller->DMSDocumentRequestForm();
+        $this->assertNotNull(
+            $form->Fields()->fieldByName('NewTextField'),
+            'DMSDocumentRequestForm() is extensible as it included the field from the extension'
+        );
+    }
+
+    /**
+     * Tests if print requests are incremented
+     */
+    public function testTrackTimestampedPrintRequest()
+    {
+        $doc = $this->objFromFixture('DMSDocument', 'doc1');
+        /** @var DMSRequestItem $item */
+        $item = DMSRequestItem::create()->setDocument($doc)->setQuantity(2);
+        $this->cart->addItem($item);
+
+        $this->controller->trackTimestampedPrintRequest();
+        $this->assertEquals(1, $item->getDocument()->PrintRequestCount);
+    }
+
+    /**
+     * Tests if a Cart is received
+     */
+    public function testGetCart()
+    {
+        $this->assertInstanceOf('DMSDocumentCart', $this->controller->getCart());
+    }
+
+    /**
+     * Tests whether the cart items are updated from the controller
+     */
+    public function testUpdateCartItems()
+    {
+        $doc = $this->objFromFixture('DMSDocument', 'doc1');
+        /** @var DMSRequestItem $item */
+        $item = DMSRequestItem::create()->setDocument($doc)->setQuantity(2);
+        $this->cart->addItem($item);
+        $updatedQuantities = array(
+            'ItemQuantity' => array($doc->ID => 5),
+        );
+        $this->controller->updateCartItems($updatedQuantities);
+        $this->assertEquals(6, $this->cart->getItem($item->getItemId())->getQuantity());
+    }
+
+    /**
+     * Tests whether the recipient details are updated from the controller
+     */
+    public function testUpdateCartReceiverInfo()
+    {
+        $newInfo = array(
+            'ReceiverName'            => 'Joe Soap',
+            'ReceiverPhone'           => '111',
+            'ReceiverEmail'           => 'joe@example.com',
+            'DeliveryAddressLine1'    => 'A1',
+            'DeliveryAddressLine2'    => 'A2',
+            'DeliveryAddressCountry'  => 'NZ',
+            'DeliveryAddressPostCode' => '6011',
+        );
+        $this->controller->updateCartReceiverInfo($newInfo);
+        $this->assertEquals($newInfo, $this->cart->getReceiverInfo());
+    }
+
+    /**
+     * Tests whether emails are sent. Emails are mocked so not actually sent.
+     */
+    public function testSend()
+    {
+        // Set admin email
+        Config::inst()->update('Email', 'admin_email', 'admin');
+        // Temporarily replace default email service
+        $page = $this->objFromFixture('DMSDocumentCartCheckoutPage', 'page1');
+        /** @var DMSDocumentCartCheckoutPage_Controller $controller */
```

Comment:
Okay. Moved $page up to setup as well as it's being repeated. So now it uses $this->controller everywhere.